arangod/Graph/KShortestPathsFinder.cpp

```diff
@@ -270,7 +270,7 @@ bool KShortestPathsFinder::computeNextShortestPath(Path& result) {
   auto& lastShortestPath = _shortestPaths.back();
   bool available = false;
 
-  for (size_t i = lastShortestPath._branchpoint; i < lastShortestPath.length() - 1; ++i) {
+  for (ssize_t i = lastShortestPath._branchpoint; i < lastShortestPath.length() - 1; ++i) {
```

Comment:
this `i` should never become negative, so if you have to satisfy a compiler check. maybe
```suggestion
  for (size_t i = lastShortestPath._branchpoint; i + 1 < lastShortestPath.length(); ++i) {
```

Reply:
The problem is that `_branchpoint` is defined to be of type `ssize_t`, so the compiler will always complain about the assignment of a signed type to an unassigned type. We either have to define `i` as the same type, or perform a cast.